### DISCUSSION TITLE
Organize files into specified paths by existing tag alone

### DISCUSSION THREAD

**Discussion by xstrex** [beetbox/beets#5126] [Q&A] [answered]
I messed something up somewhere, and now my library is all messed up, could really use some help. 

Normally I organize my music like such: `/music/artist/album/files` but my paths option got screwed up in the config file and it put all the music in `/music/artist/files` so now I have an artist directory that contains 40+ '01 - ' files, but no subdirectories. Also the files are simply named '01 - track.ext' so I've got no way to know which track goes with which album. 

Luckily the tags should all be correct, so I'm wondering if there's any way to sort the files in the `/music/artist` directories into `/music/artist/album` by tags alone? I've played with the `import` option, as well as the `move` & `update` but not sure on the right options. 

Cheers!

**Comment by wisp3rwind** [comment]
That's precisely beets usual mode of operation: When you `import` music and `move`ing files is enabled, the are moved to the locations specified via the [Path Format Configuration](https://beets.readthedocs.io/en/stable/reference/config.html#path-format-configuration). If the music is imported to beets already, the `beet move` command will move them into place.

I'm not quite following what lead to your issue, though. But as long as beets still "knows" where your files are (i.e. they are in the places where beets put them, and you didn't manually move things around), just using `beet move` after fixing up your path configuration should correct things. No need to re-import.

**Reply by xstrex** [reply, marked as answer]
Thank you, but I don't think I fully understood the problem before asking for help. An out-of-band application was adding files to my directory structure that we're being processed by beets, so it was unaware of the presence of the files, and they didn't follow the standard that I have specified in my config. 

I managed to hack together a solution, it's not elegant but it seems to be working. I figured out that I could run a `beet import -A --group-albums` and it would skip the auto-tagger (reading the tags of the files only), and group albums together based on tag alone- but needed to be pointed at a directory full of individual files. For that I just did a quick bash for loop to iterate through the directories, then called beet's to process the directories. This is what's currently running.. and sorting files into directories and based on `$album` on a per `$artist` basis. 

```bash
for i in {D..Z}*/; do beet import -A --group-albums "$i"; done
```
